File: ProgrammingBasicsJava/Jan2020/LoopsPart2Exercise/src/_03Vacation.java
import java.util.Scanner;

public class _03Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double tripExpenses = Double.parseDouble(scanner.nextLine());
        double availableMoney = Double.parseDouble(scanner.nextLine());
        int limitDaysExpenses = 5;
        int daysGone = 0;
        int spendingCounter = 0;

        while (availableMoney < tripExpenses && spendingCounter < limitDaysExpenses) {
            String action = scanner.nextLine();
            double actionedMoney = Double.parseDouble(scanner.nextLine());
            daysGone++;

            if (action.equals("spend")) {
                spendingCounter++;
                availableMoney -= actionedMoney;

                if (availableMoney < 0) {
                    availableMoney = 0;
                }
            }
            else if (action.equals("save")) {
                spendingCounter = 0;
                availableMoney += actionedMoney;
            }
            else {
                System.out.println("wrong command");
                return;
            }
        }

        if (spendingCounter == limitDaysExpenses) {
            System.out.printf("You can't save the money.%n%d", daysGone);
        }
        if (availableMoney >= tripExpenses) {
            System.out.printf("You saved the money for %d days.", daysGone);
        }
    }
}

File: JavaScriptBasics/05_2_WhileLoopExercise/03_Vacation.js
function isVacationAffordable(input) {
    let neededMoneyForVacation = Number(input[0]);
    let availableMoney = Number(input[1]);
    let index = 2;
    let daysSpending = 0;
    let days = 0

    while (availableMoney < neededMoneyForVacation && daysSpending < 5) {
        let command = input[index];
        let currentMoney = Number(input[index + 1]);
        index += 2;
        days++;

        if (command === "spend") {
            daysSpending++;
            availableMoney -= currentMoney;

            if (availableMoney < 0) {
                availableMoney = 0;
            }
        } else if (command === "save") {
            daysSpending = 0;
            availableMoney += currentMoney;
        } else {
            console.log("wrong command");
            return;
        }
    }

    if (availableMoney >= neededMoneyForVacation) {
        console.log(`You saved the money for ${days} days.`);
    } else if (daysSpending === 5) {
        console.log(`You can't save the money.\n${days}`);
    }
}

isVacationAffordable(["250",
"150",
"spend",
"50",
"spend",
"50",
"save",
"100",
"save",
"100"]);